lib/cisco_node_utils/cmd_ref/interface.yaml

```diff
@@ -0,0 +1,236 @@
+# interface
+---
+_template:
+  config_set: ["interface %s"]
+  config_get_token: '/^interface %s$/i'
+  grpc:
+    config_get: "show running interface"
+  nxapi:
+    config_get: "show running interface all"
+
+access_vlan:
+  config_get_token_append: '/^switchport access vlan (.*)$/'
+  config_set_append: "switchport access vlan %s"
+  default_value: 1
+
+admin_state_ethernet_noswitchport_shutdown:
+  # TODO: is this actually used?
+  nxapi:
+    /N7K/:
+      default_value: "shutdown"
+
+all_interfaces:
+  config_get_token: '/^interface (.*)/'
+
+create:
+  config_set: "interface %s"
+
+description:
+  config_get_token_append: '/^description (.*)/'
+  config_set_append: "%s description %s"
+  default_value: ""
+
+destroy:
+  config_set: "no interface %s"
+
+encapsulation_dot1q:
+  config_get_token_append: '/^encapsulation dot1q (.*)/'
+  config_set_append: "%s encapsulation dot1q %s"
+  default_value: ""
+
+feature_lacp:
+  config_get: "show running | i ^feature"
+  config_get_token: '/^feature lacp$/'
+  config_set: "%s feature lacp"
+
+feature_vlan:
+  config_get: "show running | i ^feature"
+  config_get_token: '/^feature interface-vlan$/'
+  config_set: "%s feature interface-vlan"
+
+ipv4_addr_mask:
+  config_get_token_append: '/^ip address ([0-9\.]+)[\s\/](.*)/'
+  config_set_append: "%s ip address %s"
+
+ipv4_address:
+  default_value: ~
+
+ipv4_netmask_length:
+  default_value: ~
+
+ipv4_proxy_arp:
+  config_get_token_append: '/^ip proxy-arp$/'
+  config_set_append: "%s ip proxy-arp"
+  default_value: false
+
+ipv4_redirects_loopback:
+  nxapi:
+    /N7K/:
+      default_value: false
+      config_set: ~
+      config_get: ~
+      config_get_token: ~
+    else:
+      config_get_token_append: '/^((?:no )?ip redirects)$/'
+      config_set_append: "%s ip redirects"
+      default_value: true
+  test_config_get_regex: [!ruby/regexp '/^\s+no ip redirects/']
+  test_config_result:
+    false: RuntimeError
+    true: RuntimeError
+
+ipv4_redirects_other_interfaces:
+  config_get_token_append: '/^((?:no )?ip redirects)$/'
+  config_set_append: "%s ip redirects"
+  default_value: true
+  test_config_get_regex: [!ruby/regexp '/^\s+no ip redirects/']
+  test_config_result:
+    false: false
+    true: true
+
+mtu:
+  config_get_token_append: '/^mtu (.*)$/'
+  config_set_append: "%s mtu %s"
+  default_value: 1500
+
+negotiate_auto_ethernet:
+  /(N7K|C3064)/:
+    config_get: ~
+    config_set: ~
+    config_get_token: ~
+    default_value: false
+  else:
+    config_get_token_append: '/^negotiate auto$/'
+    config_set_append: "%s negotiate auto"
+    default_value: true
+  test_config_get_regex: [
+  !ruby/regexp '/^\s+no negotiate auto/',
```

Comment:
It was still required for test_config_get_regex (only) but I've fixed that - new diffs incoming.